user-contracts-mon/internal/monitor.go

```diff
@@ -0,0 +1,175 @@
+package monitor
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/NicoNex/echotron/v3"
+	"github.com/threefoldtech/tfgrid-sdk-go/grid-client/deployer"
+	"github.com/threefoldtech/tfgrid-sdk-go/grid-client/graphql"
+)
+
+// Monitor struct of parsed configration
+type Monitor struct {
+	Bot      echotron.API
+	Mnemonic string
+	BotToken string
+	Network  string
+	interval int
+}
+
+// NewMonitor creates a new monitor from parsed config/env file
+func NewMonitor(conf Config) Monitor {
+	mon := Monitor{}
+	mon.Mnemonic = conf.mnemonic
+	mon.Network = conf.network
+	mon.interval = conf.interval
+	mon.BotToken = conf.botToken
+	mon.Bot = echotron.NewAPI(mon.BotToken)
+
+	return mon
+}
+
+// StartMonitoring starts monitoring the contracts with
+// specific mnemonics and notify subscribed chats every fixed interval
+func (mon Monitor) StartMonitoring(tfPluginClient deployer.TFPluginClient, addChatChan chan int64, stopChatChan chan int64) {
+	chatIDs := map[int64]bool{}
+	ticker := time.NewTicker(time.Duration(mon.interval) * time.Hour)
+
+	for {
+		select {
+		case chatID := <-addChatChan:
+			chatIDs[chatID] = true
+
+		case chatID := <-stopChatChan:
+			chatIDs[chatID] = false
+
+		case <-ticker.C:
+			contractsInGracePeriod, contractsAgainstDownNodes, err := runMonitor(tfPluginClient)
```

Comment:
I am not sure the bot will be working with multiple chat IDs, only one right? so no need for a map. also I am not sure if the stop command should just stop the bot from sending messages or stop it from doing anything, like finding contracts. what do you think?